### PR TITLE
REC: Silicon tracking

### PR DESCRIPTION
**Pull request by fucd**
* Add RecAssociationMaker
* Add ITrackerFitterTool
  * change silicon tracking fitter interface to KalTestTool
* Add TStraightTrack case into TKalDetCradle while B=0
* fix bugs

### CODE REVIEW

File: Reconstruction/SiliconTracking/src/SiliconTrackingAlg.cpp [hunk 1]
```diff
@@ -2793,11 +2864,13 @@ void SiliconTrackingAlg::FinalRefit(edm4hep::TrackCollection* trk_col) {
 	error() << "Cannot create MarlinTrack ! " << endmsg;
 	return;
       }
+      */
```
Comment (mirguest) [inline Reconstruction/SiliconTracking/src/SiliconTrackingAlg.cpp]: Hi @fucd, 
Is this part not used any more? If it is not used, please consdier to remove them.
Tao

Comment (fucd) [reply Reconstruction/SiliconTracking/src/SiliconTrackingAlg.cpp]: Yes, I have removed them.

**Review by mirguest — APPROVED** [review]

**Comment by mirguest** [comment]
Hi @fucd,
This PR looks good to me. Do you have any other changes? Please let me know, then I can merge your PR. 
Thank you. 
Tao

**Comment by fucd** [comment]
Hi @mirguest 

Thanks very much. I have no more update in this PR, and for geometry, I will create anothe PR.

Chengdong

**Pull request merged**